examples/external_aerodynamics/data_sources.py

```diff
@@ -232,20 +233,21 @@ def _write_zarr(
             output_path: Path where the .zarr directory should be written
         """
         # Create store
-        zarr_store = zarr.DirectoryStore(output_path)
-        root = zarr.group(store=zarr_store)
+        zarr_store = LocalStore(output_path)
+        root = zarr.open_group(store=zarr_store, mode="w")
 
         # Write metadata as attributes
+        data.metadata.zarr_format = zarr.__version__
         root.attrs.update(asdict(data.metadata))
 
         # Write required arrays
         for field in ["stl_coordinates", "stl_centers", "stl_faces", "stl_areas"]:
             array_info = getattr(data, field)
-            root.create_dataset(
-                field,
+            root.create_array(
```

Comment:
Sharding could go here:
```
shards = array_info.shards
```
presumably.  I am not sure where array_info is made, maybe ~10 to ~100 chunks per shard?